presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -904,6 +935,22 @@ public void dropTable(MetastoreContext metastoreContext, String databaseName, St
         }
     }
 
+    private static void deleteDirectoryRecursively(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
+    {
+        try {
+            hdfsEnvironment.getFileSystem(context, path).delete(path, true);
+        }
+        catch (IOException | RuntimeException e) {
+            // don't fail if unable to delete path
+            log.warn(e, "Failed to delete path: " + path.toString());
+        }
+    }
+
+    private static boolean isManagedTable(Table table)
+    {
+        return table.getTableType().equals(MANAGED_TABLE.name());
+    }
+
```

Comment:
I see this is same code as in GlueHiveMetastore
https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java#L545-L559

wondering if we can move these methods to MetastoreUtil and use in both places instead of writing the same code twice.

Reply:
@NikhilCollooru I have moved them to MetastoreUtil, let me know if you have any more suggestions.